src/components/chips/js/chipController.js

```diff
@@ -0,0 +1,181 @@
+   * @type {$timeout}
+   */
+  this.$timeout = $timeout;
+
+  /**
+   * @type {boolean}
+   */
+  this.isEditting = false;
+
+  /**
+   * @type {boolean}
+   */
+  this.enableChipEdit = $scope.$parent && $scope.$parent.enableChipEdit == 'true';
+
+  if(this.enableChipEdit) {
+    this.$element.on('keydown', this.chipKeyDown.bind(this));
+    this.$element.on('dblclick', this.doubleClicked.bind(this));
+  }
+}
+
+
+/**
+ * @return {angular.JQLite}
+ */
+MdChipCtrl.prototype.getChipContent = function() {
+  var chipContents = this.$element[0].getElementsByClassName('md-chip-content')
+  return angular.element(chipContents[0]);
+};
+
+
+/**
+ * @return {angular.JQLite}
+ */
+MdChipCtrl.prototype.getContentElement = function() {
+  return angular.element(this.getChipContent().children()[0]);
+};
+
+
+/**
+ * @return {angular.JQLite}
+ */
+MdChipCtrl.prototype.getEditInput = function() {
+  return angular.element(this.getChipContent().find('input')[0]);
+};
+
+
+/**
+ * Presents an input element to edit the contents of the chip.
+ */
+MdChipCtrl.prototype.goOutOfEditMode = function() {
+  this.isEditting = false;
+  this.$element.removeClass('md-chip-editing');
+  var currentText = this.getEditInput().val();
+  this.getContentElement().text(currentText);
+  this.getEditInput().remove();
+  this.getContentElement().removeClass('ng-hide');
+};
+
+
+/**
+ * Measures the width of a text in given font.
+ * @param text
+ * @param font
+ */
+MdChipCtrl.prototype.getTextWidth = function(text, font) {
+    // re-use canvas object for better performance
+    var canvas = document.createElement("canvas");
+    var context = canvas.getContext("2d");
+    context.font = font;
+    var metrics = context.measureText(text);
+    return metrics.width;
+};
+
+/**
+ * Get's the computed style of element for given CSS property.
+ * @param element
+ * @param property
+ */
+MdChipCtrl.prototype.getComputedStyle = function(element, property) {
+    if(typeof window.getComputedStyle !== 'function') {
+        return undefined;
+    }
+
+    return window.getComputedStyle( element, null ).getPropertyValue( property );
+};
+
+/**
+ * Adjusts the input width to match its contents
+ */
+MdChipCtrl.prototype.adjustEditInputWidth = function() {
+  var currentText = this.getEditInput().val();
+  var fontSize = this.getComputedStyle(this.getEditInput()[0], 'font-size') || '16px';
+  var fontFamily = this.getComputedStyle(this.getEditInput()[0], 'font-family') || 'Roboto';
+  var width = this.getTextWidth(currentText, fontSize + ' ' + fontFamily);
+  this.getEditInput().css('width', width + 'px');
+};
+
+
+/**
+ * Presents an input element to edit the contents of the chip.
+ */
+MdChipCtrl.prototype.goInEditMode = function() {
+  this.isEditting = true;
+  this.$element.addClass('md-chip-editing');
+  var currentText = this.getContentElement().text();
+  this.getContentElement().addClass('ng-hide');
+  this.getChipContent().append('<input type="text" value="'+currentText+'" />');
```

Comment:
Just thought a bit about the input. It's just an idea, can we not use a contend editable div? This will work really good and doesn't need all the text measures. Then we just need to watch for the content editable blur and update the actual chip model.